MaxText/train.py

```diff
@@ -310,10 +312,13 @@ def train_loop(config, state=None):
       max_utils.activate_profiler(config)
 
     example_batch = load_next_batch(data_iterator, example_batch, config)
+    step_start_time = datetime.datetime.now()
     with mesh, nn_partitioning.axis_rules(config.logical_axis_rules):
       state, metrics, nextrng = p_train_step(
           state, example_batch, nextrng
       )
+    step_stop_time = datetime.datetime.now()
```

Comment:
This code doesn't do what you think it does -- Jax calls are non blocking. I think it would better to provide a goodput.goodput_log_step_time call that took step time (and inside of it you can get a datetime).